cubed/runtime/executors/spark.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Optional, Sequence
+
+from networkx import MultiDiGraph
+from pyspark.sql import SparkSession
+
+from cubed.runtime.pipeline import visit_nodes
+from cubed.runtime.types import Callback, DagExecutor
+from cubed.runtime.utils import handle_callbacks, handle_operation_start_callbacks
+from cubed.spec import Spec
+
+
+class SparkExecutor(DagExecutor):
+    """An execution engine that uses Apache Spark."""
+
+    # Minimum memory allowed for Spark (512MB)
+    MIN_MEMORY_MB = 512
+
+    def __init__(self, **kwargs):
+        self._spark_session = None
+        self._lazy_operations = []
+        self._callbacks = None
+        super().__init__(**kwargs)
+
+    @property
+    def name(self) -> str:
+        return "spark"
+
+    def _parse_memory_setting(self, memory_value):
+        """
+        Convert memory value to a valid Spark memory setting string.
+        Ensures the value is at least MIN_MEMORY_MB.
+
+        Args:
+            memory_value: Memory value (can be int bytes or string like '1g')
+
+        Returns:
+            String memory setting suitable for Spark config (e.g., '512m')
+        """
+        # If it's already a string with a unit, return as is
```

Comment:
The value from `spec.allowed_mem` is [always an int](https://github.com/cubed-dev/cubed/blob/2b39036de3c3398480f9cfa7ff2f4523470e7fdd/cubed/spec.py#L61) representing the number of bytes, so you don't need to do this check. You could also annotate `memory_value` in this function as an `int`.